src/cpp/src/llm_pipeline_static.cpp

```diff
@@ -408,38 +655,51 @@ void StaticLLMPipeline::setupAndCompileModels(
         1) Read the template model - this will be kvcache model
         2) Expose KV-cache input and output layers from kvcache model
         3) Align u4 ZP constants - TODO: get rid of this step in future
-        4) Replace KV-cache tensors for the entire cache to tensors only for new token (before concat)
-        5) Clone the model - this will be prefill
-        6) Reshape both models to static shape
-        7) Compile both models
+        4) Clone the model - this will be prefill
+        5) Reshape both models to static shape
+        6) Apply layout optimization if applicable
+        7) Replace KV-cache tensors for the entire cache to tensors only for new token (before concat)
+        8) Convert kv-cache tensors to f16 precision
+        9) Compile both models
     */
 
     ov::Core core;
 
     // NB: Get information about NPU if available
     auto npudesc = extract_npu_descriptor(core);
-
     // (1) Read the template model - this will be kvcache model
     m_kvcache_model = core.read_model((models_path / "openvino_model.xml").string());
     // (2) Expose KV-cache input and output layers from kvcache model
     ov::pass::StatefulToStateless().run_on_model(m_kvcache_model);
     // (3) Align u4 ZP constants
     align_u4_zp_constants(m_kvcache_model);
-    // (4) Replace KV-tensors for the entire cache to tensors only for new token
-    m_kvcache_model = redirect_new_kv_to_output(m_kvcache_model);
-    // (5) Convert kvcache tensors to fp16 precision
-    m_kvcache_model = cvt_kvcache_to_fp16(m_kvcache_model);
-    // (6) Clone the model - this will be prefill
+    // (4) Clone the model - this will be prefill
     m_prefill_model = m_kvcache_model->clone();
     m_prefill_model->set_friendly_name(m_kvcache_model->get_friendly_name() + "_prefill");
-    // (7) Reshape both models to static shape
+    // (5) Reshape both models to static shape
     const uint32_t kMaxPromptLen = align_to(pop_int_and_cast(properties, "MAX_PROMPT_LEN").value_or(1024u), 64u);
     const uint32_t kMinResponseLen = align_to(pop_int_and_cast(properties, "MIN_RESPONSE_LEN").value_or(128u), 64u);
-    KVAxesPosition axes = get_kv_axes(get_model_type_from_json(models_path / "config.json"));
-    m_kvcache_desc = KVCacheDesc { kMaxPromptLen, kMaxPromptLen + kMinResponseLen, 0u, axes.seq_len };
+    ModelDesc model_desc = get_modeldesc_from_json(models_path / "config.json");
+    KVAxesPosition axes = get_kv_axes(model_desc.type);
+    m_kvcache_desc = KVCacheDesc { kMaxPromptLen, kMaxPromptLen + kMinResponseLen, 0u, axes.seq_len, false};
     reshape_to_static(m_prefill_model, m_kvcache_desc.max_prompt_size, m_kvcache_desc.max_prompt_size, axes);
     reshape_to_static(m_kvcache_model, 1u, m_kvcache_desc.total_size, axes);
-    // (8) Compile both model
+    // (6) Apply opt layout if applicable
+    // NB: Try to apply opt transpose only for Llama-2-7b-chat-hf model
+    if ( model_desc.name_or_path == "meta-llama/Llama-2-7b-chat-hf" ||
```

Comment:
I think both lhs & rhs in `==` should be lowercased first to make comparison more robust

Reply:
Will create a ticket for this!

Reply:
do we need hardcode for specific model? is not `model_desc.type == "llama"` enough ?

Reply:
@ilya-lavrenov we're not confident (yet) if it will work for other llama versions. @TolyaTalamanov will generalize this change as he gets back from vacation